contracts/liquidity_hub/bonding-manager/src/helpers.rs

```diff
@@ -0,0 +1,260 @@
+        }));
+    }
+    Ok(())
+}
+
+// Used in FillRewards to search the funds for coins that are neither LP tokens nor whale and swap them to whale
+pub fn swap_coins_to_main_token(
+    info: MessageInfo,
+    deps: &DepsMut,
+    config: white_whale_std::bonding_manager::Config,
+    whale: &mut Coin,
+    distribution_denom: &String,
+    messages: &mut Vec<CosmosMsg>,
+) -> Result<(), ContractError> {
+    let coins_to_swap: Vec<&Coin> = info
+        .funds
+        .iter()
+        .filter(|coin| {
+            !coin.denom.contains(".pair.")
+                & !coin.denom.contains(LP_SYMBOL)
+                & !coin.denom.eq(distribution_denom)
+        })
+        .collect();
+    for coin in coins_to_swap {
+        let swap_route_query = white_whale_std::pool_manager::QueryMsg::SwapRoute {
+            offer_asset_denom: coin.denom.to_string(),
+            ask_asset_denom: distribution_denom.to_string(),
+        };
+
+        // Query for the routes and pool
+        let swap_routes: SwapRouteResponse = deps
+            .querier
+            .query_wasm_smart(config.pool_manager_addr.to_string(), &swap_route_query)?;
+
+        // check if the pool has any assets, if not skip the swap
+        // Note we are only checking the first operation here. Might be better to another loop to check all operations
+        let pool_query = white_whale_std::pool_manager::QueryMsg::Pair {
+            pair_identifier: swap_routes
+                .swap_route
+                .swap_operations
+                .first()
+                .unwrap()
+                .get_pool_identifer(),
+        };
+        let resp: PairInfoResponse = deps
+            .querier
+            .query_wasm_smart(config.pool_manager_addr.to_string(), &pool_query)?;
+        let mut skip_swap = false;
+        // Check pair 'assets' and if either one has 0 amount then don't do swaps
+        resp.pair_info.assets.iter().for_each(|asset| {
+            if asset.amount.is_zero() {
+                skip_swap = true;
+            }
+        });
+
+        let simulate: SimulateSwapOperationsResponse = deps.querier.query_wasm_smart(
+            config.pool_manager_addr.to_string(),
+            &white_whale_std::pool_manager::QueryMsg::SimulateSwapOperations {
+                offer_amount: coin.amount,
+                operations: swap_routes.swap_route.swap_operations.clone(),
+            },
+        )?;
+        // Add the simulate amount received to the whale amount, if the swap fails this should also be rolled back
+        whale.amount = whale.amount.checked_add(simulate.amount)?;
+
+        if !skip_swap {
+            // 1% max spread for the swap
+            let msg = white_whale_std::pool_manager::ExecuteMsg::ExecuteSwapOperations {
+                operations: swap_routes.swap_route.swap_operations.clone(),
+                minimum_receive: Some(simulate.amount),
+                to: None,
+                max_spread: Some(Decimal::percent(1)),
```

Comment:
I think here we want to swap regardless, so we might pump this slippage percentage or not?